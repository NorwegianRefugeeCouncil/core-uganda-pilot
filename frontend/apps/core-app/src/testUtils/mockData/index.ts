export { makeRecord } from './Record';
export { makeForm } from './Form';
export { makeField } from './FormField';
