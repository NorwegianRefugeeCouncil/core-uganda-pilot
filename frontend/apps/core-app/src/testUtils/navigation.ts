export const mockNavigationProp = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
};
