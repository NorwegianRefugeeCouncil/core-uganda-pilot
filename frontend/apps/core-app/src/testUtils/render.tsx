import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  render as r,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import { render as rWeb } from '@testing-library/react';
import { NativeBaseProvider } from 'native-base';
import { theme } from 'core-design-system';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const render = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
): RenderAPI =>
  r(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );

export const renderWeb = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
) =>
  rWeb(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );
