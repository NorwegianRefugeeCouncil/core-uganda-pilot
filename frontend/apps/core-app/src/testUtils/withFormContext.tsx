import { Button } from 'native-base';
import React from 'react';
import { FieldValues, FormProvider, useForm } from 'react-hook-form';

export const withFormContext =
  <T,>(
    Component: React.FC<T>,
    defaultValues: FieldValues = {},
    onSubmit: (data: any) => void = () => {},
  ) =>
  // eslint-disable-next-line react/display-name
  (props: T) => {
    const f = useForm({ defaultValues });

    const handleSubmit = f.handleSubmit((data: any) => {
      onSubmit(data);
    });

    return (
      <FormProvider {...f}>
        <Component {...props} />
        <Button testID="with-form-context-submit-button" onPress={handleSubmit}>
          Submit
        </Button>
      </FormProvider>
    );
  };
