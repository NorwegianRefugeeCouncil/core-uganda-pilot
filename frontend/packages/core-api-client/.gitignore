/lib
/node_modules

#testing
/coverage
