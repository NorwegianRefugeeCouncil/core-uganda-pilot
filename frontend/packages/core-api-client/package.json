{
  "name": "core-api-client",
  "version": "0.1.0",
  "description": "> TODO: description",
  "homepage": "https://github.com/nrc-no/core#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/nrc-no/core.git"
  },
  "bugs": {
    "url": "https://github.com/nrc-no/core/issues"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1"
  }
}
