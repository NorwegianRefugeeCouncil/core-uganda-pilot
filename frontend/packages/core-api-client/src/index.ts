export { Client } from './lib/client';
export { BaseRESTClient } from './lib/client/BaseRESTClient';
export * from './lib/types';
export * from './lib/utils';
