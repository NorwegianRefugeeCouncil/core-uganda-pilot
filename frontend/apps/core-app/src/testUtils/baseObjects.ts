import { FormDefinition, FormType, Record } from 'core-api-client';

export const baseForm: FormDefinition = {
  id: 'form1',
  code: '',
  name: 'form 1',
  databaseId: 'databaseId1',
  folderId: 'folderId1',
  formType: FormType.DefaultFormType,
  fields: [],
};

export const baseTextField = {
  id: 'field1',
  name: 'field 1',
  description: 'description 1',
  code: '',
  required: false,
  key: false,
  fieldType: {
    text: {},
  },
};

export const baseMultilineTextField = {
  id: 'field2',
  name: 'field 2',
  description: 'description 2',
  code: '',
  required: true,
  key: true,
  fieldType: {
    multilineText: {},
  },
};

export const baseRecord1: Record = {
  id: 'record1',
  values: [
    {
      value: 'text',
      fieldId: 'field1',
    },
    {
      value: 'multi line text',
      fieldId: 'field2',
    },
  ],
  formId: 'form1',
  databaseId: 'databaseId1',
  ownerId: undefined,
};
