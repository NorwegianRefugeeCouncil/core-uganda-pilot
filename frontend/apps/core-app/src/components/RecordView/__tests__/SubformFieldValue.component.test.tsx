import { FieldKind } from 'core-api-client';

import { render } from '../../../testUtils/render';
import { SubformFieldValueComponent } from '../SubformFieldValue.component';

it('should match the snapshot', () => {
  const { toJSON } = render(
    <SubformFieldValueComponent
      header="header"
      labels={['label-1', 'label-2']}
      items={[
        [
          {
            fieldType: FieldKind.Text,
            value: 'test-1',
            formattedValue: 'format-test-1',
          },
          {
            fieldType: FieldKind.Reference,
            value: 'test-1',
            formattedValue: 'format-test-1',
          },
        ],
        [
          {
            fieldType: FieldKind.Text,
            value: 'test-2',
            formattedValue: 'format-test-2',
          },
          {
            fieldType: FieldKind.Reference,
            value: 'test-2',
            formattedValue: 'format-test-2',
          },
        ],
      ]}
    />,
  );
  expect(toJSON()).toMatchSnapshot();
});
