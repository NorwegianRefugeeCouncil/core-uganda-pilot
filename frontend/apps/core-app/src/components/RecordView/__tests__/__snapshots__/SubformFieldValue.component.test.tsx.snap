// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={Object {}}
  >
    <Text
      dataSet={Object {}}
      level="3"
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#000000",
          "fontFamily": "Roboto_500Medium",
          "fontSize": 24,
          "fontStyle": undefined,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 28,
          "textAlign": "auto",
          "textDecorationLine": undefined,
        }
      }
    >
      header
    </Text>
    <View
      dataSet={Object {}}
      style={Object {}}
    />
    <View
      dataSet={Object {}}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={Object {}}
      >
        <Text
          dataSet={Object {}}
          level="1"
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#666666",
              "fontFamily": "Roboto_400Regular",
              "fontSize": 18,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 26,
              "textAlign": "auto",
              "textDecorationLine": undefined,
            }
          }
        >
          label-1
        </Text>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <Text
          dataSet={Object {}}
          level="1"
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#000000",
              "fontFamily": "Roboto_500Medium",
              "fontSize": 18,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 26,
              "textAlign": "auto",
              "textDecorationLine": undefined,
            }
          }
        >
          format-test-1
        </Text>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <Text
          dataSet={Object {}}
          level="1"
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#000000",
              "fontFamily": "Roboto_500Medium",
              "fontSize": 18,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 26,
              "textAlign": "auto",
              "textDecorationLine": undefined,
            }
          }
        >
          format-test-2
        </Text>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "width": 12,
          }
        }
      />
      <View
        dataSet={Object {}}
        style={Object {}}
      >
        <Text
          dataSet={Object {}}
          level="1"
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#666666",
              "fontFamily": "Roboto_400Regular",
              "fontSize": 18,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 26,
              "textAlign": "auto",
              "textDecorationLine": undefined,
            }
          }
        >
          label-2
        </Text>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="link"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "height": "auto",
              "width": "auto",
            }
          }
        >
          <Text
            dataSet={Object {}}
            level="1"
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#000000",
                "fontFamily": "Roboto_400Regular",
                "fontSize": 20,
                "fontStyle": undefined,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 34,
                "textAlign": "auto",
                "textDecorationLine": "underline",
              }
            }
          >
            format-test-1
          </Text>
        </View>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <View
          accessibilityRole="link"
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "height": "auto",
              "width": "auto",
            }
          }
        >
          <Text
            dataSet={Object {}}
            level="1"
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#000000",
                "fontFamily": "Roboto_400Regular",
                "fontSize": 20,
                "fontStyle": undefined,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 34,
                "textAlign": "auto",
                "textDecorationLine": "underline",
              }
            }
          >
            format-test-2
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
